sqlmesh/core/test/definition.py

```diff
@@ -485,6 +484,49 @@ def generate_test(
         yaml.dump({test_name: test_body}, file)
 
 
+def _create_df(
+    rows: t.List[Row], columns: t.Optional[t.Collection] = None, partial: t.Optional[bool] = False
+) -> pd.DataFrame:
+    if columns:
+        referenced_columns = {col for row in rows for col in row}
+        _raise_if_unknown_columns(columns, referenced_columns)
+
+        if partial:
+            columns = list(referenced_columns)
+
+    return pd.DataFrame.from_records(rows, columns=columns)
+
+
+def _raise_if_unknown_columns(
```

Comment:
would `_raise_if_extra_columns` make more sense? Based on the name I was expecting a check for `UNKNOWN` data type.

Reply:
Yeah, I can see the confusion. Maybe an even better name could be `_raise_if_unexpected_columns`?